src/porepy/models/contact_mechanics_model.py

```diff
@@ -583,10 +497,12 @@ def _assign_discretizations(self) -> None:
         Assign discretizations to the nodes and edges of the grid bucket.
 
         """
+
         # For the Nd domain we solve linear elasticity with mpsa.
         Nd = self._Nd
         gb = self.gb
-
+        if not hasattr(self, "dof_manager"):
+            self.dof_manager = pp.DofManager(gb)
```

Comment:
Honestly, I'm not sure

Reply:
This will be safe (unless someone changes the GridBucket, but we can live with that). Iteration over dictionaries, which is the data structure GridBucket uses to store its subdomains and interfaces, will be in a deterministic order.